spec/policies/exercise_policy_spec.rb

```diff
@@ -122,6 +122,28 @@
     end
   end
 
+  %i[submit?].each do |action|
```

Comment:
This looks suspicious. Tests seem to be very specific to the action. Why do the `each` stuff then?
I didn't look into details.

Reply:
Thanks for having a look. I simplified the test cases by removing the iteration with the `each` wherever possible. This has been done for all permission checks in a dedicated commit.